custom_model_runner/datarobot_drum/drum/push.py

```diff
@@ -3,60 +3,30 @@
 from pathlib import Path
 
 import datarobot as dr_client
-from strictyaml import Bool, Int, load, Map, Optional, Str, YAMLError
 
-from datarobot_drum.drum.common import RunMode
+from datarobot_drum.drum.common import RunMode, MODEL_CONFIG_FILENAME, TargetType
 from datarobot_drum.drum.exceptions import DrumCommonException
 
-CONFIG_FILENAME = "model-metadata.yaml"
 DR_LINK_FORMAT = "{}/model-registry/custom-models/{}"
 MODEL_LOGS_LINK_FORMAT = "{url}/projects/{project_id}/models/{model_id}/log"
 
-schema = Map(
-    {
-        "name": Str(),
-        "type": Str(),
-        "environmentID": Str(),
-        "targetType": Str(),
-        "validation": Map({"input": Str(), Optional("targetName"): Str()}),
-        Optional("modelID"): Str(),
-        Optional("description"): Str(),
-        Optional("majorVersion"): Bool(),
-        Optional("inferenceModel"): Map(
-            {
-                "targetName": Str(),
-                Optional("positiveClassLabel"): Str(),
-                Optional("negativeClassLabel"): Str(),
-                Optional("predictionThreshold"): Int(),
-            }
-        ),
-        Optional("trainingModel"): Map({Optional("trainOnProject"): Str()}),
-    }
-)
-
-
-def _read_metadata(code_dir):
-    code_dir = Path(code_dir)
-    if not code_dir.joinpath(CONFIG_FILENAME).exists():
+
+def _get_metadata(options):
+    code_dir = Path(options.code_dir)
+    if options.model_config is None:
         raise DrumCommonException(
             "You must have a file with the name {} in the directory {}. \n"
             "You don't. \nWhat you do have is these files: \n{} ".format(
-                CONFIG_FILENAME, code_dir, os.listdir(code_dir)
+                MODEL_CONFIG_FILENAME, code_dir, os.listdir(code_dir)
             )
         )
-    with open(code_dir.joinpath(CONFIG_FILENAME)) as f:
-        try:
-            model_config = load(f.read(), schema).data
-        except YAMLError as e:
-            print(e)
-            raise SystemExit()
-    return model_config
+    return options.model_config
 
 
 def _convert_target_type(unconverted_target_type):
-    if unconverted_target_type == "regression":
+    if unconverted_target_type == TargetType.REGRESSION.value:
         return dr_client.TARGET_TYPE.REGRESSION
-    elif unconverted_target_type == "binary":
+    elif unconverted_target_type == TargetType.BINARY.value:
```

Comment:
@jlpolit did you add anomaly to drum push? doesn't look like it?

Reply:
I did not - it's going to require some changes to the api client, which I have a PR open for but was told to hold off until they published a version....which reminds me I should really check on that

Reply:
(this is that thing https://github.com/datarobot/public_api_client/pull/1307 )